frontend/src/container/Trace/TraceGraphFilter/index.tsx

```diff
@@ -38,19 +53,86 @@ function TraceGraphFilter(): JSX.Element {
 	};
 
 	const onClickSelectedGroupByHandler = (ev: unknown): void => {
-		if (typeof ev === 'string') {
-			const selected = groupBy.find((e) => e.key === ev);
+		if (typeof ev === 'string' && options) {
+			const selected = options.find((e) => e.value === ev);
 			if (selected) {
 				dispatch({
 					type: UPDATE_SELECTED_GROUP_BY,
 					payload: {
-						selectedGroupBy: selected.key,
+						selectedGroupBy: selected.value ? selected.value.toString() : '',
 					},
 				});
 			}
 		}
 	};
+	function groupByValues(
+		tagFilters: {
+			value: string;
+			label: string;
+		}[],
+	): {
+		value: string;
+		label: string;
+	}[] {
+		const result: { value: string; label: string }[] = tagFilters;
+		groupBy.forEach((e) => {
+			result.push({
+				value: e.key,
+				label: e.displayValue,
+			});
+		});
+		return result;
+	}
+	const onSearchHandler = useCallback(async () => {
+		try {
+			setSelectLoading(true);
+			const response = await getTagFilters({
+				start: globalTime.minTime,
+				end: globalTime.maxTime,
+				other: Object.fromEntries(traces.selectedFilter),
+				isFilterExclude: traces.isFilterExclude,
+			});
 
+			if (response.statusCode === 200) {
+				if (response.payload === null) {
+					setOptions(
+						groupBy.map((e) => ({
+							value: e.key,
+							label: e.displayValue,
+						})),
+					);
+				} else {
+					setOptions(
+						groupByValues(
+							extractTagFilters(response.payload).map((e) => ({
```

Comment:
can we shift the logic of map in `extractTagFilters`

Reply:
No, as `extractTagFilters` is only expected to return array

Reply:
Moved to utils